client/my-sites/site-settings/form-seo.jsx

```diff
@@ -0,0 +1,279 @@
+	},
+
+	submitSeoForm( event ) {
+		const site = this.props.site;
+
+		if ( ! event.isDefaultPrevented() && event.nativeEvent ) {
+			event.preventDefault();
+		}
+
+		notices.clearNotices( 'notices' );
+
+		this.setState( { submittingForm: true } );
+
+		const filteredCodes = omitBy( this.state.verificationServicesCodes, isBoolean );
+		const updatedOptions = {
+			seo_meta_description: this.state.seoMetaDescription,
+			verification_services_codes: filteredCodes
+		};
+
+		site.saveSettings( updatedOptions, error => {
+			if ( error ) {
+				switch ( error.error ) {
+					case 'invalid_ip':
+						notices.error( this.translate( 'One of your IP Addresses was invalid. Please, try again.' ) );
+						break;
+					default:
+						notices.error( this.translate( 'There was a problem saving your changes. Please, try again.' ) );
+				}
+				this.setState( { submittingForm: false } );
+			} else {
+				notices.success( this.translate( 'Settings saved!' ) );
+				this.markSaved();
+				this.setState( { submittingForm: false } );
+
+				site.fetchSettings();
+			}
+		} );
+
+		analytics.tracks.recordEvent( 'calypso_seo_settings_form_submit' );
+	},
+
+	getGeneralSettingsUrl( slug ) {
+		return `/settings/general/${ slug }`;
+	},
+
+	render() {
+		const { slug = '', settings: { blog_public = 1 } = {}, fetchingSettings } = this.props.site;
+		const { submittingForm, seoMetaDescription, verificationServicesCodes } = this.state;
+
+		const isSitePrivate = parseInt( blog_public, 10 ) !== 1;
+		const isDisabled = isSitePrivate || fetchingSettings || submittingForm;
+		const isMetaError = seoMetaDescription && seoMetaDescription.length > 160;
+
+		const sitemapUrl = `https://${ slug }/sitemap.xml`;
+		const generalSettingsUrl = this.getGeneralSettingsUrl( slug );
+
+		// The API returns 'false' for an empty array value, so we force it to an empty string if needed
+		const googleCode = get( verificationServicesCodes, 'google' ) || '';
+		const bingCode = get( verificationServicesCodes, 'bing' ) || '';
+		const pinterestCode = get( verificationServicesCodes, 'pinterest' ) || '';
+		const yandexCode = get( verificationServicesCodes, 'yandex' ) || '';
+
+		return (
+			<div className={ this.state.fetchingSettings ? 'is-loading' : '' }>
+				<PageViewTracker path="/settings/seo/:site" title="Site Settings > SEO" />
+				{ isSitePrivate &&
+					<Notice status="is-warning" showDismiss={ false } text={ this.translate( 'SEO settings are disabled because the site visibility is not set to Public.' ) }>
+						<NoticeAction href={ generalSettingsUrl }>{ this.translate( 'View Settings' ) }</NoticeAction>
+					</Notice>
+				}
+				<SectionHeader label={ this.translate( 'Search Engine Optimization' ) }>
+					<Button
+						compact={ true }
+						onClick={ this.submitSeoForm }
+						primary={ true }
+						type="submit"
+						disabled={ isDisabled || this.state.submittingForm }
+					>
+						{ this.state.submittingForm
```

Comment:
we're already pulling in this variable from `this.state`, so you don't need to repeat that here

also maybe we should consider renaming it `isSubmittingForm` in the destructuring.